core/lib/zksync_core/src/metadata_calculator/mod.rs

```diff
@@ -191,10 +191,12 @@ impl MetadataCalculator {
         events_queue_commitment: Option<H256>,
         bootloader_initial_content_commitment: Option<H256>,
     ) -> L1BatchMetadata {
-        let is_pre_boojum = header
+        // The commitment generation pre-boojum is the same for all the version, so in case the version is not present, we just supply the
+        // last pre-boojum version.
+        // TODO(X): make sure that protocol version is not an Option
```

Comment:
is this needed?

Reply:
Same as here: https://github.com/matter-labs/zksync-era/pull/791#discussion_r1447282310